### PR TITLE
Gvrf controls sample new features

### PR DESCRIPTION
**Pull request by Maginador**
Controls Sample update adding customization menu with animation features and touch pad representation for input demonstration (as done with gamepad).

GearVRf-DCO-1.0-Signed-off-by: Ricardo Maginador
ricardo.cm@samsung.com

### CODE REVIEW

**Comment by dknaan** [comment]
I tried no Note4 (Mali) and on S6.
In both of them the application starts, I see gray, brown, and blue (grass, fence, and sky?) for 1 second and then the application exist.

When I check Android Runtime I get the following:
D/AndroidRuntime(13276): AndroidRuntime START com.android.internal.os.RuntimeInit <<<<<<
D/AndroidRuntime(13276): CheckJNI is OFF
D/AndroidRuntime(13276): propertySet: couldn't set property (it is from app)
D/AndroidRuntime(13276): propertySet: couldn't set property (it is from app)
D/AndroidRuntime(13276): propertySet: couldn't set property (it is from app)
D/AndroidRuntime(13276): propertySet: couldn't set property (it is from app)
D/AndroidRuntime(13276): propertySet: couldn't set property (it is from app)
D/AndroidRuntime(13276): readGMSProperty: start
D/AndroidRuntime(13276): readGMSProperty: already setted!!
D/AndroidRuntime(13276): propertySet: couldn't set property (it is from app)
D/AndroidRuntime(13276): readGMSProperty: could not set the property(default)!!
D/AndroidRuntime(13276): readGMSProperty: end
D/AndroidRuntime(13276): addProductProperty: start
D/AndroidRuntime(13276): Calling main entry com.android.commands.am.Am
D/AndroidRuntime(13276): Shutting down VM

**Comment by nathan-almeida** [comment]
Fixed.
Please run it again.

**Comment by dknaan** [comment]
Cool app!

I checked it on my Mali Note4 and S6 with Oculus 0.6.0.1. It works fine most of the time, but:
* The application sometimes crashed immediately when I insert it to my headset (please see log below).
* The application crashed when I press the 'back' button of the GearVr.
* When I rotated the the star in the setting screen it didn't affect the orientation of the star in the application (the scale, color, motion worked well and affected the app).
* I didn't check the code. Just used the application. There are many changes so please let me know if there is anything specific to check.

Following is the log with the error:

D/AndroidRuntime(32355): >>>>>> AndroidRuntime START com.android.internal.os.RuntimeInit <<<<<<
D/AndroidRuntime(32355): CheckJNI is ON
D/AndroidRuntime(32355): readGMSProperty: start
D/AndroidRuntime(32355): readGMSProperty: already setted!!
D/AndroidRuntime(32355): readGMSProperty: end
D/AndroidRuntime(32355): addProductProperty: start
D/AndroidRuntime(32355): Calling main entry com.android.commands.am.Am
D/AndroidRuntime(32355): Shutting down VM
D/AndroidRuntime(32474): Shutting down VM
E/AndroidRuntime(32474): FATAL EXCEPTION: main
E/AndroidRuntime(32474): Process: org.gearvrf.controls, PID: 32474
E/AndroidRuntime(32474): java.lang.NullPointerException: Attempt to invoke virtual method 'float android.view.MotionEvent.getX()' on a null object reference
E/AndroidRuntime(32474): 	at org.gearvrf.controls.util.VRSamplesTouchPadGesturesDetector.onScroll(VRSamplesTouchPadGesturesDetector.java:146)
E/AndroidRuntime(32474): 	at android.view.GestureDetector.onTouchEvent(GestureDetector.java:588)
E/AndroidRuntime(32474): 	at android.support.v4.view.GestureDetectorCompat$GestureDetectorCompatImplJellybeanMr2.onTouchEvent(GestureDetectorCompat.java:475)
E/AndroidRuntime(32474): 	at android.support.v4.view.GestureDetectorCompat.onTouchEvent(GestureDetectorCompat.java:538)
E/AndroidRuntime(32474): 	at org.gearvrf.controls.util.VRSamplesTouchPadGesturesDetector.onTouchEvent(VRSamplesTouchPadGesturesDetector.java:129)
E/AndroidRuntime(32474): 	at org.gearvrf.controls.MainActivity.onTouchEvent(MainActivity.java:93)
E/AndroidRuntime(32474): 	at org.gearvrf.GVRActivity.dispatchTouchEvent(GVRActivity.java:269)
E/AndroidRuntime(32474): 	at com.android.internal.policy.impl.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:2529)
E/AndroidRuntime(32474): 	at android.view.View.dispatchPointerEvent(View.java:9284)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4767)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4605)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4129)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4182)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4148)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4262)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4156)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4319)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4129)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4182)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4148)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4156)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4129)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:6627)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:6517)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:6488)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:6717)
E/AndroidRuntime(32474): 	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:185)
E/AndroidRuntime(32474): 	at android.view.InputEventReceiver.nativeConsumeBatchedInputEvents(Native Method)
E/AndroidRuntime(32474): 	at android.view.InputEventReceiver.consumeBatchedInputEvents(InputEventReceiver.java:176)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl.doConsumeBatchedInput(ViewRootImpl.java:6688)
E/AndroidRuntime(32474): 	at android.view.ViewRootImpl$ConsumeBatchedInputRunnable.run(ViewRootImpl.java:6740)
E/AndroidRuntime(32474): 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:777)
E/AndroidRuntime(32474): 	at android.view.Choreographer.doCallbacks(Choreographer.java:590)
E/AndroidRuntime(32474): 	at android.view.Choreographer.doFrame(Choreographer.java:558)
E/AndroidRuntime(32474): 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:763)
E/AndroidRuntime(32474): 	at android.os.Handler.handleCallback(Handler.java:739)
E/AndroidRuntime(32474): 	at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime(32474): 	at android.os.Looper.loop(Looper.java:145)
E/AndroidRuntime(32474): 	at android.app.ActivityThread.main(ActivityThread.java:5946)
E/AndroidRuntime(32474): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(32474): 	at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(32474): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1388)
E/AndroidRuntime(32474): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1183)
E/AndroidRuntime(32474): Error reporting crash
E/AndroidRuntime(32474): android.os.DeadObjectException
E/AndroidRuntime(32474): 	at android.os.BinderProxy.transactNative(Native Method)
E/AndroidRuntime(32474): 	at android.os.BinderProxy.transact(Binder.java:503)
E/AndroidRuntime(32474): 	at android.app.ActivityManagerProxy.handleApplicationCrash(ActivityManagerNative.java:4666)
E/AndroidRuntime(32474): 	at com.android.internal.os.RuntimeInit$UncaughtHandler.uncaughtException(RuntimeInit.java:95)
E/AndroidRuntime(32474): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:693)
E/AndroidRuntime(32474): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:690)
D/AndroidRuntime( 1410): Shutting down VM
E/AndroidRuntime( 1410): FATAL EXCEPTION: main
E/AndroidRuntime( 1410): Process: com.samsung.android.hmt.vrsvc, PID: 1410
E/AndroidRuntime( 1410): java.lang.RuntimeException: Unable to create service com.samsung.android.hmt.vrsvc.service.HmtService: java.lang.RuntimeException: Package manager has died
E/AndroidRuntime( 1410): 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:3161)
E/AndroidRuntime( 1410): 	at android.app.ActivityThread.access$1900(ActivityThread.java:178)
E/AndroidRuntime( 1410): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1532)
E/AndroidRuntime( 1410): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime( 1410): 	at android.os.Looper.loop(Looper.java:145)
E/AndroidRuntime( 1410): 	at android.app.ActivityThread.main(ActivityThread.java:5946)
E/AndroidRuntime( 1410): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 1410): 	at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime( 1410): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1388)
E/AndroidRuntime( 1410): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1183)
E/AndroidRuntime( 1410): Caused by: java.lang.RuntimeException: Package manager has died
E/AndroidRuntime( 1410): 	at android.app.ApplicationPackageManager.getPackageInfo(ApplicationPackageManager.java:134)
E/AndroidRuntime( 1410): 	at dj.Qxpw(Util.java:1053)
E/AndroidRuntime( 1410): 	at com.samsung.android.hmt.vrsvc.service.HmtService.onCreate(HmtService.java:196)
E/AndroidRuntime( 1410): 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:3151)
E/AndroidRuntime( 1410): 	... 9 more
E/AndroidRuntime( 1410): Caused by: android.os.DeadObjectException
E/AndroidRuntime( 1410): 	at android.os.BinderProxy.transactNative(Native Method)
E/AndroidRuntime( 1410): 	at android.os.BinderProxy.transact(Binder.java:503)
E/AndroidRuntime( 1410): 	at android.content.pm.IPackageManager$Stub$Proxy.getPackageInfo(IPackageManager.java:1961)
E/AndroidRuntime( 1410): 	at android.app.ApplicationPackageManager.getPackageInfo(ApplicationPackageManager.java:129)
E/AndroidRuntime( 1410): 	... 12 more
E/AndroidRuntime( 1410): Error reporting crash
E/AndroidRuntime( 1410): android.os.DeadObjectException
E/AndroidRuntime( 1410): 	at android.os.BinderProxy.transactNative(Native Method)
E/AndroidRuntime( 1410): 	at android.os.BinderProxy.transact(Binder.java:503)
E/AndroidRuntime( 1410): 	at android.app.ActivityManagerProxy.handleApplicationCrash(ActivityManagerNative.java:4666)
E/AndroidRuntime( 1410): 	at com.android.internal.os.RuntimeInit$UncaughtHandler.uncaughtException(RuntimeInit.java:95)
E/AndroidRuntime( 1410): 	at cw.uncaughtException(VRServiceApplication.java:29)
E/AndroidRuntime( 1410): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:693)
E/AndroidRuntime( 1410): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:690)
E/AndroidRuntime(29963): FATAL EXCEPTION: libraries-thread
E/AndroidRuntime(29963): Process: com.android.vending, PID: 29963
E/AndroidRuntime(29963): java.lang.RuntimeException: android.os.DeadObjectException
E/AndroidRuntime(29963): 	at android.accounts.AccountManager.getAccountsByTypeAsUser(AccountManager.java:500)
E/AndroidRuntime(29963): 	at android.accounts.AccountManager.getAccountsByType(AccountManager.java:491)
E/AndroidRuntime(29963): 	at com.google.android.finsky.api.AccountHandler.getAccounts(AccountHandler.java:26)
E/AndroidRuntime(29963): 	at com.google.android.finsky.FinskyApp$2.getAccounts(FinskyApp.java:406)
E/AndroidRuntime(29963): 	at com.google.android.finsky.appstate.AppStatesReplicator.bucketAppsByOwner(AppStatesReplicator.java:382)
E/AndroidRuntime(29963): 	at com.google.android.finsky.appstate.AppStatesReplicator.internalReplicate(AppStatesReplicator.java:146)
E/AndroidRuntime(29963): 	at com.google.android.finsky.appstate.AppStatesReplicator.access$100(AppStatesReplicator.java:52)
E/AndroidRuntime(29963): 	at com.google.android.finsky.appstate.AppStatesReplicator$2.run(AppStatesReplicator.java:123)
E/AndroidRuntime(29963): 	at android.os.Handler.handleCallback(Handler.java:739)
E/AndroidRuntime(29963): 	at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime(29963): 	at android.os.Looper.loop(Looper.java:145)
E/AndroidRuntime(29963): 	at android.os.HandlerThread.run(HandlerThread.java:61)
E/AndroidRuntime(29963): Caused by: android.os.DeadObjectException
E/AndroidRuntime(29963): 	at android.os.BinderProxy.transactNative(Native Method)
E/AndroidRuntime(29963): 	at android.os.BinderProxy.transact(Binder.java:503)
E/AndroidRuntime(29963): 	at android.accounts.IAccountManager$Stub$Proxy.getAccountsAsUser(IAccountManager.java:745)
E/AndroidRuntime(29963): 	at android.accounts.AccountManager.getAccountsByTypeAsUser(AccountManager.java:497)
E/AndroidRuntime(29963): 	... 11 more
E/AndroidRuntime(29963): Error reporting crash
E/AndroidRuntime(29963): android.os.DeadObjectException
E/AndroidRuntime(29963): 	at android.os.BinderProxy.transactNative(Native Method)
E/AndroidRuntime(29963): 	at android.os.BinderProxy.transact(Binder.java:503)
E/AndroidRuntime(29963): 	at android.app.ActivityManagerProxy.handleApplicationCrash(ActivityManagerNative.java:4666)
E/AndroidRuntime(29963): 	at com.android.internal.os.RuntimeInit$UncaughtHandler.uncaughtException(RuntimeInit.java:95)
E/AndroidRuntime(29963): 	at com.google.android.finsky.FinskyApp$CrashHandler.uncaughtException(FinskyApp.java:287)
E/AndroidRuntime(29963): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:693)
E/AndroidRuntime(29963): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:690)

**Comment by chenchao1407** [comment]
Try to run with Note4(Adreno) and S6(Mali). It looks smooth and fine with S6. But with Note4, I can see significant frame rate down and also, I can see textures looks abnormal with the scene objects like game pad and the tea pot(? not sure what it is).

**Comment by nathan-almeida** [comment]
Crashes have been fixed;
Rotate star, along with some new features are working in progress and will be delivered in a few days;

We now have an issue after we press Back and click on 'No' to return to application. Apparently many textures are not loaded and are just replaced automatically by some texts. If you can take a look at this, it will be of great help for us. We also have seen that solar system sample has the same issue.

I don't have an answer for the issue of frame rate dropping.

**Comment by thomasflynn** [comment]
looks like it needs a rebase.

**Comment by nathan-almeida** [comment]
Just to clarify:

We don`t have the issue of frame rate dropping in our devices. We have tested in Note4(Mali). Apparently this is a performance issue with snapdragon's CPUs.

Our code is currently updated with the latest version of Framework, and the issue with textures not being loaded still happens. Maybe we could open an issue for that?

@thomasflynn: I didn't quite get your comment. Could you be a little more specific?

**Comment by thomasflynn** [comment]
the button to merge your change is grayed out.  it says, "This branch has conflicts that must be resolved".  i cannot merge it to master.

**Comment by Maginador** [comment]
Here it says "Only those with write access to this repository can merge pull requests."
Is there anything we can or have to do?

**Comment by thomasflynn** [comment]
Yes.  you need to do a 'git rebase' and rebase your change.

**Comment by Maginador** [comment]
Updated pull request branch.

File: GVRf/Sample/gvrf_controls_sample/assets/gvr_note4.xml [hunk 1]
```diff
@@ -38,10 +38,10 @@
             colorFormat="COLOR_8888"
             depthFormat="DEPTH_24"
             fov-y="90.0"
-            multiSamples="1"
+            multiSamples="8"
```
Comment (thomasflynn) [inline GVRf/Sample/gvrf_controls_sample/assets/gvr_note4.xml]: i should probably already know the answer to this, but do any of the current mobile chipsets support 8xMSAA?  what happens if you ask for 8xMSAA and it doesn't exist?

File: GVRf/Sample/gvrf_controls_sample/res/values/strings.xml [hunk 2]
```diff
@@ -6,7 +6,7 @@
  *
  *     http://www.apache.org/licenses/LICENSE-2.0
  *
- * Unless required by applicable law or agreed to in writing, software
+ * Unless required by applicable law or agreed to in writing, softwares
```
Comment (thomasflynn) [inline GVRf/Sample/gvrf_controls_sample/res/values/strings.xml]: fwiw, "software" was correct.  Software is both singular and plural.  English is weird :)

**Comment by thomasflynn** [comment]
looks good.  i have some minor comments, but not a reason to hold up the check-in.  if you want to address them, you can do so in a separate pull-request.

**Pull request merged**